src/Main/Main.js

```diff
@@ -13,17 +13,27 @@ export class Main extends Component {
   componentDidMount() {
     this.props.fetchFromFirebase();
     this.props.getUserLocation();
-
   }
 
-  componentWillUpdate(nextProps) {
-    if ( this.props.competitions.length !== nextProps.competitions.length) {
-      const { activeUser, competitions } =  nextProps;
-      loadUserComps(activeUser, competitions);
+  shouldComponentUpdate(nextProps, nextState) {
+    if (this.props.liveUser &&
+      (this.props.competitions.length !== nextProps.competitions.length)) {
+      this.props.loadUserComps(this.props.activeUser, nextProps.competitions);
     }
+    return true;
   }
 
   render() {
+    let displayComps;
+    if (this.props.location.pathname === '/my-competitions') {
+      this.props.competitions ?
```

Comment:
solid refactoring of the conditional out of the render. Could possibly work the ternary here into the if statement
let displayComps = [];
if (this.props.location.pathname === '/my-competitions' && this.props.competitions) { 
displayComps = this.props.activeUser.competitions;
}

Reply:
ooo i likey

Reply:
I had this written like three or four different ways, and this way gave me the least amount of bugs so I stuck with it :-)